plugins/inmemoryscanner/src/Scanner.cpp

```diff
@@ -91,18 +118,22 @@ namespace InMemoryScanner
                     pluginInterface->logMessage(LogLevel::debug,
                                                 LOG_FILENAME,
                                                 "Start dumpVadRegionToFile with size: " +
-                                                    intToHex(memoryRegion->size()));
-                    dumping->dumpMemoryRegion(processName, pid, memoryRegionDescriptor, *memoryRegion);
+                                                    intToHex(memoryMapping->getSizeInGuest()));
+
+                    auto paddedRegion = constructPaddedMemoryRegion(*mappedRegions);
+
+                    dumping->dumpMemoryRegion(processName, pid, memoryRegionDescriptor, paddedRegion);
                     pluginInterface->logMessage(LogLevel::debug, LOG_FILENAME, "End dumpVadRegionToFile");
                 }
 
-                pluginInterface->logMessage(
-                    LogLevel::debug, LOG_FILENAME, "Start scanMemory with size: " + intToHex(memoryRegion->size()));
+                pluginInterface->logMessage(LogLevel::debug,
+                                            LOG_FILENAME,
+                                            "Start scanMemory with size: " + intToHex(memoryMapping->getSizeInGuest()));
 
                 // The semaphore protects the yara rules from being accessed more than YR_MAX_THREADS (32 atm.) times in
                 // parallel.
                 semaphore.wait();
-                auto results = yaraEngine->scanMemory(*memoryRegion);
+                auto results = yaraEngine->scanMemory(*mappedRegions);
```

Comment:
Iirc we previously tried to scan all regions in parallel. Do you remember why we decided to scan the processes in parallel instead?